src/views/Generator/GeneratorTabs/RequestList.tsx

```diff
@@ -0,0 +1,107 @@
+import { Button, Flex, ScrollArea } from '@radix-ui/themes'
+import { useShallowCompareEffect } from 'react-use'
+
+import { WebLogView } from '@/components/WebLogView'
+import { ProxyData } from '@/types'
+import { Filter } from '@/components/WebLogView/Filter'
+import { useFilterRequests } from '@/components/WebLogView/Filter.hooks'
+import { useProxyDataGroups } from '@/hooks/useProxyDataGroups'
+import { useStudioUIStore } from '@/store/ui'
+import { useGeneratorStore } from '@/store/generator'
+import { EmptyMessage } from '@/components/EmptyMessage'
+import { GlobeIcon } from '@radix-ui/react-icons'
+
+interface RequestListProps {
+  requests: ProxyData[]
+  onSelectRequest: (request: ProxyData | null) => void
+  selectedRequest: ProxyData | null
+}
+
+export function RequestList({
+  requests,
+  onSelectRequest,
+  selectedRequest,
+}: RequestListProps) {
+  const {
+    filter,
+    setFilter,
+    filteredRequests,
+    filterAllData,
+    setFilterAllData,
+  } = useFilterRequests({
+    proxyData: requests,
+  })
+
+  const groups = useProxyDataGroups(requests)
+
+  const recordings = useStudioUIStore((state) => state.recordings)
+  const recordingPath = useGeneratorStore((state) => state.recordingPath)
+  const allowlist = useGeneratorStore((store) => store.allowlist)
+
+  const setShowAllowlistDialog = useGeneratorStore(
+    (store) => store.setShowAllowlistDialog
+  )
+
+  const recording = recordings.get(recordingPath)
+
+  const isRecordingMissing = recording === undefined && recordingPath !== ''
+  const areHostsSelected = allowlist.length > 0
+
+  // Preserve the selected request when modifying rules
+  useShallowCompareEffect(() => {
+    onSelectRequest(null)
+  }, [requests])
+
+  return (
+    <Flex direction="column" height="100%">
+      <ScrollArea scrollbars="vertical">
+        {isRecordingMissing && (
+          <Flex justify="center" align="center" height="100%">
+            <EmptyMessage
+              message="The selected recording is missing, select another one from the top menu"
+              pt="0"
+            />
+          </Flex>
+        )}
+        {!areHostsSelected && (
```

Comment:
There are at least 5 different cases when we might want to show a message like this, maybe we can address some of them with the same message if it's generic enough: 
1. No recording is selected (if the user creates an empty generator)
2. Selected recording missing
3. Selected recording is empty (an edge case, but still)
4. Allowlist is empty
5. Allowlist is not empty, but the selected hosts are only used for static assets (see example below)

<img width="737" alt="image" src="https://github.com/user-attachments/assets/3be65426-49cc-4021-9d27-39db9f00c432" />